calico/getting-started/kubernetes/managed-public-cloud/aks-migrate.mdx

```diff
@@ -0,0 +1,139 @@
+
+  ```bash
+  az aks show --name my-calico-cluster --resource-group my-calico-rg  \
+    --query "networkProfile.networkPlugin, networkProfile.networkPolicy"
+[
+  "azure",
+  "calico"
+]
+  ```
+
+:::warning
+Application connectivity may be disrupted during the migration
+
+Network policy will be removed during the migration and must be re-applied after the migration.
+:::
+
+
+1. Get all existing network policies from all namespaces and write them out to a file
+    ```bash
+    kubectl get netpol -A -o yaml  > network-policy.yaml
+    ```
+    Review the file to check everything you expect to be there, is.
+
+1. Do the migration
+    ```bash
+    az aks update --name my-calico-cluster --resource-group my-calico-rg --network-policy=none
+    ```
+
+1. Confirm that the migration has completed successfully
+    ```bash
+    az aks show --name my-calico-cluster --resource-group my-calico-rg  \
+      --query "networkProfile.networkPlugin, networkProfile.networkPolicy"
+    [
+    "azure",
+    "none"
+    ]
+    ```
+
+1. Install $[prodname].
+    Note that we do not install CRDs here, since they may already exist. Operator will check and create any missing ones.
+    ```bash
+    kubectl create -f https://raw.githubusercontent.com/projectcalico/calico/v3.30.3/manifests/tigera-operator.yaml
+    ```
+
+1. Configure the $[prodname] installation
+    ```bash
+    kubectl create -f - <<EOF
+    kind: Installation
+    apiVersion: operator.tigera.io/v1
+    metadata:
+      name: default
+    spec:
+      kubernetesProvider: AKS
+      cni:
+        type: AzureVNET
+    ---
+
+    # This optional section configures the Calico API server.
+    # For more information, see: https://docs.tigera.io/calico/latest/reference/installation/api#operator.tigera.io/v1.APIServer
+    apiVersion: operator.tigera.io/v1
+    kind: APIServer
+    metadata:
+      name: default
+    spec: {}
+
+    ---
+
+    # This optional section enables the Calico Goldmane flow aggregator.
+    apiVersion: operator.tigera.io/v1
+    kind: Goldmane
+    metadata:
+      name: default
+
+    ---
+
+    # This optional section enables the Calico Whisker observability UI.
+    apiVersion: operator.tigera.io/v1
+    kind: Whisker
+    metadata:
+      name: default
+    EOF
+    ```
+
+1. Confirm that $[prodname] is up and running
+    ```bash
+    kubectl get tigerastatus
+
+    NAME        AVAILABLE   PROGRESSING   DEGRADED   SINCE
+    apiserver   True        False         False      100s
+    calico      True        False         False      50s
+    goldmane    True        False         False      55s
+    ippools     True        False         False      105s
+    whisker     True        False         False      80s
+    ```
+Check that all components report `AVAILABLE` as `True`.
+
+1. Re-apply all existing policies
+    ```bash
+    kubectl apply -f network-policy.yaml
+    ```
```

Comment:
Is there a good verification that this is right?

Reply:
Not really.  If it doesn't succeed, there will be an error?

Reply:
I guess you could run `kubectl get networkpolicy -A -o yaml` and eyeball the output?

Reply:
Nah let's leave it for now.